src/core/lib/event_engine/windows/grpc_polled_fd_windows.cc

```diff
@@ -224,7 +224,7 @@ class GrpcPolledFdWindows : public GrpcPolledFd {
     // c-ares overloads this recv_from virtual socket function to receive
     // data on both UDP and TCP sockets, and from is nullptr for TCP.
     if (from != nullptr) {
-      CHECK(*from_len <= recv_from_source_addr_len_);
```

Comment:
This change itself looks right to me, but I'm now confused how this was ever working.

`recv_from_source_addr_len_` should always be 200 (`recv_from_source_addr_` is statically allocated 200 bytes).

Working on that assumption, regardless of old vs. new c-ares, I think `*from_len` should always be less than `recv_from_source_addr_len_`. 

The comment in https://github.com/grpc/grpc/issues/37969#issuecomment-2435340438 shows `recv_from_source_addr_len_` is `16`. So I think we have a bug in how we're setting `recv_from_source_addr_len_`.

Reply:
I think when we call [WSARecvFrom](https://github.com/grpc/grpc/blob/8b7496172de3fe9fc221b7299ac42fd8b6ad546c/src/core/resolver/dns/c_ares/grpc_ares_ev_driver_windows.cc#L188-L191), we passed `recv_from_source_addr_len_` as an input and output argument. So it will be modified on completion to indicate the actual size of the address.

Reply:
Whoops! Thank you, I misread that.

In that case, since we're definitely relying on `WSARecvFrom` to populate `recv_from_source_addr_len_`, how about just initializing `recv_from_source_addr_len_` to zero, instead of `sizeof(recv_from_source_addr_)` ?

We might then also add a `CHECK(recv_from_source_addr_len_ != 0)` right before this `from_len` check.

Reply:
done.

Reply:
oh sorry, it's an input and output argument, so its initial value (`sizeof(recv_from_source_addr_)`) is useful to indicate the size of `recv_from_source_addr_`. So I will revert back this.